jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/EntrypointIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.maven;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class EntrypointIntegrationTest {
+
+  @ClassRule public static final TestPlugin testPlugin = new TestPlugin();
+
+  @ClassRule
+  public static final TestProject entrypointTestProject = new TestProject(testPlugin, "entrypoint");
+
+  /**
+   * Builds and runs jib:buildTar on a project at {@code projectRoot} and examines the resulting
+   * configuration.
+   */
+  @Test
+  public void testExecute() throws VerificationException, IOException {
+    String targetImage = "entrypoint:maven" + System.nanoTime();
+
+    Verifier verifier = new Verifier(entrypointTestProject.getProjectRoot().toString());
+    verifier.setSystemProperty("_TARGET_IMAGE", targetImage);
+    verifier.setAutoclean(false);
+    verifier.executeGoal("package");
+
+    verifier.executeGoal("jib:" + BuildTarMojo.GOAL_NAME);
```

Comment:
It looks like this should be part of `BuildTarMojoIntegrationTest`?

Reply:
Okay, let's add a javadoc describing what this class is testing/

Reply:
But we are using `jib:buildTar` to build a tar and then verifying that it has the correct entrypoint here, aren't we? I think I agree that it does make more sense to put this in `BuildTarMojoIntegrationTest`. As far as I understand, the test file should correspond directly to a source file (and we don't have an Entrypoint class in jib-maven-plugin).